src/base/RequestClient.ts

```diff
@@ -245,27 +297,56 @@ class RequestClient {
     this.lastResponse = undefined;
     this.lastRequest = new Request(requestOptions);
 
-    return this.axios(options)
-      .then((response) => {
-        if (opts.logLevel === "debug") {
-          console.log(`response.statusCode: ${response.status}`);
-          console.log(`response.headers: ${JSON.stringify(response.headers)}`);
-        }
-        _this.lastResponse = new Response(
-          response.status,
-          response.data,
-          response.headers
+    try {
+      let response: globalThis.Response;
+      
+      if (this.autoRetry) {
+        response = await performRequestWithRetry(
+          this.fetch,
+          url,
+          fetchOptions,
+          {
+            maxIntervalMillis: this.maxRetryDelay,
+            maxRetries: this.maxRetries,
+          }
         );
-        return {
-          statusCode: response.status,
-          body: response.data,
-          headers: response.headers,
-        };
-      })
-      .catch((error) => {
-        _this.lastResponse = undefined;
-        throw error;
-      });
+      } else {
+        response = await this.fetch(url, fetchOptions);
+      }
+
+      if (opts.logLevel === "debug") {
+        console.log(`response.statusCode: ${response.status}`);
```

Comment:
It might be worth to also log response.statusText, especially if (!response.ok)
It might also be worth logging more info in case of an error, like:

    if (!response.ok) {
      const error = await response.json();
      console.log ` blah blah ... ${error} ...`
      }